scheduler/src/cook/mesos/pool.clj

```diff
@@ -1,6 +1,16 @@
 (ns cook.mesos.pool
   (:require [datomic.api :as d]))
 
+(defn check-pool
+  "Returns true if requesting-default-pool? and the entity does not have a pool
+   or the entity has a pool named pool-name"
+  [source eid relationship pool-name requesting-default-pool?]
```

Comment:
Maybe `entity->pool-fn` instead of `relationship`?

Reply:
I like `entity->pool` if that works for you